contrib/babelfishpg_tsql/src/multidb.c

```diff
@@ -1324,116 +1325,117 @@ get_physical_user_name(char *db_name, char *user_name, bool suppress_error)
+
+	if ((0 == strcmp(dbname, "master")) || (0 == strcmp(dbname, "tempdb")) || (0 == strcmp(dbname, "msdb")))
+	{	
+		snprintf(name, MAX_BBF_NAMEDATALEND, "%s_dbo", dbname);
+	}
+	else if (SINGLE_DB == get_migration_mode())
+	{	
+		snprintf(name, MAX_BBF_NAMEDATALEND, "%s", "dbo");
+	}
 	else
 	{
-		char	   *name = palloc0(MAX_BBF_NAMEDATALEND);
-
 		snprintf(name, MAX_BBF_NAMEDATALEND, "%s_dbo", dbname);
 		truncate_identifier(name, strlen(name), false);
-		return name;
 	}
+	return name;
 }
 
-const char *
+char *
 get_db_owner_name(const char *dbname)
 {
-	if (0 == strcmp(dbname, "master"))
-		return "master_db_owner";
-	if (0 == strcmp(dbname, "tempdb"))
-		return "tempdb_db_owner";
-	if (0 == strcmp(dbname, "msdb"))
-		return "msdb_db_owner";
-	if (SINGLE_DB == get_migration_mode())
-		return "db_owner";
+	char	   *name = palloc0(MAX_BBF_NAMEDATALEND);
+
+	if ((0 == strcmp(dbname, "master")) || (0 == strcmp(dbname, "tempdb")) || (0 == strcmp(dbname, "msdb")))
+	{	
+		snprintf(name, MAX_BBF_NAMEDATALEND, "%s_db_owner", dbname);
+	}
+	else if (SINGLE_DB == get_migration_mode())
+	{	
+		snprintf(name, MAX_BBF_NAMEDATALEND, "%s", "db_owner");
+	}
 	else
 	{
-		char	   *name = palloc0(MAX_BBF_NAMEDATALEND);
-
 		snprintf(name, MAX_BBF_NAMEDATALEND, "%s_db_owner", dbname);
 		truncate_identifier(name, strlen(name), false);
-		return name;
 	}
+	return name;
 }
 
-const char *
+char *
 get_guest_role_name(const char *dbname)
 {
-	if (0 == strcmp(dbname, "master"))
-		return "master_guest";
-	if (0 == strcmp(dbname, "tempdb"))
-		return "tempdb_guest";
-	if (0 == strcmp(dbname, "msdb"))
-		return "msdb_guest";
+	char	   *name = palloc0(MAX_BBF_NAMEDATALEND);
 
 	/*
 	 * Always prefix with dbname regardless if single or multidb. Note that
 	 * dbo is an exception.
 	 */
-	else
-	{
-		char	   *name = palloc0(MAX_BBF_NAMEDATALEND);
-
-		snprintf(name, MAX_BBF_NAMEDATALEND, "%s_guest", dbname);
-		truncate_identifier(name, strlen(name), false);
-		return name;
-	}
+	snprintf(name, MAX_BBF_NAMEDATALEND, "%s_guest", dbname);
+	truncate_identifier(name, strlen(name), false);
+	return name;
 }
 
-const char *
+char *
 get_guest_schema_name(const char *dbname)
 {
-	if (0 == strcmp(dbname, "master"))
-		return "master_guest";
-	if (0 == strcmp(dbname, "tempdb"))
-		return "tempdb_guest";
-	if (0 == strcmp(dbname, "msdb"))
-		return "msdb_guest";
+	char	   *name = palloc0(MAX_BBF_NAMEDATALEND);
 
-	if (SINGLE_DB == get_migration_mode())
-		return "guest";
+	if ((0 == strcmp(dbname, "master")) || (0 == strcmp(dbname, "tempdb")) || (0 == strcmp(dbname, "msdb")))
+	{	
+		snprintf(name, MAX_BBF_NAMEDATALEND, "%s_guest", dbname);
+	}
+	else if (SINGLE_DB == get_migration_mode())
+	{	
+		snprintf(name, MAX_BBF_NAMEDATALEND, "%s", "guest");
+	}
 	else
 	{
-		char	   *name = palloc0(MAX_BBF_NAMEDATALEND);
-
 		snprintf(name, MAX_BBF_NAMEDATALEND, "%s_guest", dbname);
 		truncate_identifier(name, strlen(name), false);
-		return name;
 	}
+	return name;
```

Comment:
First and last condition can be in the same block.

Reply:
I don't think so because then the function would return `dbo` for `master`, `tempdb` and `msdb` also.

Reply:
I meant this can be just two if conditions.
```
if (single db && not any built int db)
   snprintf(name, MAX_BBF_NAMEDATALEND, "%s", "guest");
else
   snprintf(name, MAX_BBF_NAMEDATALEND, "%s_guest", dbname);
   truncate_identifier(name, strlen(name), false);
  ```